api/v1beta1/common.go

```diff
@@ -102,3 +104,25 @@ type NetworksSection struct {
 	// FixedIP - Specific IP address to use for this network
 	FixedIP string `json:"fixedIP,omitempty" yaml:"fixedIP,omitempty"`
 }
+
+// UniqueSpecFields - the array of fields that must be unique between role and nodes
+var UniqueSpecFields = []string{"OpenStackAnsibleEERunnerImage", "NetworkAttachments"}
+
+// AssertUniquenessBetween - compare specs for uniqueness
+func AssertUniquenessBetween(spec interface{}, otherSpec interface{}, suffix string) []string {
+	vSpec := reflect.ValueOf(spec)
+	vOtherSpec := reflect.ValueOf(otherSpec)
+
+	var errorMsgs []string
+	for _, field := range UniqueSpecFields {
+		value := vSpec.FieldByName(field)
+		otherValue := vOtherSpec.FieldByName(field)
+		if value.IsZero() || otherValue.IsZero() {
+			continue
+		}
+		if !reflect.DeepEqual(value.Interface(), otherValue.Interface()) {
+			errorMsgs = append(errorMsgs, field+" mismatch between "+suffix)
```

Comment:
Nice!

Reply:
@SeanMooney I agree that we want more test coverage

Can you give me an example of how to better implement `AssertUniquenessBetween`? Something which uses 
https://go.dev/doc/tutorial/generics ?

Reply:
I've filed this issue: https://github.com/openstack-k8s-operators/dataplane-operator/issues/114